casper/src/main/scala/coop/rchain/casper/util/comm/ApproveBlockProtocol.scala

```diff
@@ -0,0 +1,131 @@
+package coop.rchain.casper.util.comm
+
+import com.google.protobuf.ByteString
+
+import cats.Monad
+import cats.effect.Sync
+import cats.effect.concurrent.Ref
+import cats.implicits._
+
+import coop.rchain.casper.{PrettyPrinter, Validate}
+import coop.rchain.casper.protocol._
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.hash.Blake2b256
+import coop.rchain.comm.CommError.ErrorHandler
+import coop.rchain.comm.PeerNode
+import coop.rchain.comm.discovery.NodeDiscovery
+import coop.rchain.comm.protocol.rchain.Packet
+import coop.rchain.comm.transport
+import coop.rchain.comm.transport.TransportLayer
+import coop.rchain.shared._
+import coop.rchain.catscontrib.{Capture, IOUtil}
+
+import monix.execution.atomic.AtomicAny
+
+import scala.util.Try
+
+class ApproveBlockProtocol[
+    F[_]: Capture: Sync: NodeDiscovery: TransportLayer: Log: Time: ErrorHandler] private (
```

Comment:
`Capture` and `Sync` are the same things.

Reply:
Ah I see, `Sync.delay` is the same as `Capture.capture`. Cool, feel free to remove usage of `Capture` in favour of `Sync` then.